### PR TITLE
TGP-2285: Bug fix for url manipulation error

### PR DESCRIPTION
**Pull request by yohan-oc**

### CODE REVIEW

File: app/controllers/AssessmentController.scala [hunk 1]
```diff
@@ -162,8 +188,26 @@ class AssessmentController @Inject() (
...
+            handleDataCleansingAndRecovery(
+              request.userAnswers.id,
+              routes.CategorisationPreparationController.startLongerCategorisation(mode, recordId)
+            )
+          )
+        )
+        .recover(_ =>
+          handleDataCleansingAndRecovery(
+            request.userAnswers.id,
+            routes.CategorisationPreparationController.startLongerCategorisation(mode, recordId)
```
Comment (ChrisCookOC) [inline app/controllers/AssessmentController.scala]: Should not go here. Should go to normal startCategorisation. You will delete the data for longer categorisation when you do the data cleansing for Categorisation Journey

**Review by ChrisCookOC — CHANGES_REQUESTED** [review]

File: app/controllers/AssessmentController.scala (hunk 1 above)
Comment (yohan-oc) [reply app/controllers/AssessmentController.scala]: in both onPageLoadReassessment and onSubmitReassessment right?

Comment (ChrisCookOC) [reply app/controllers/AssessmentController.scala]: Yeah whenever we journey recovery :)

Comment (yohan-oc) [reply app/controllers/AssessmentController.scala]: Updated

**Review by ChrisCookOC — APPROVED** [review]

**Pull request merged**